src/business_logic/transaction/transactions.rs

```diff
@@ -0,0 +1,45 @@
+use crate::{
+    business_logic::{
+        execution::objects::TransactionExecutionInfo,
+        state::state_api::{State, StateReader},
+    },
+    definitions::general_config::StarknetGeneralConfig,
+    utils::Address,
+};
+
+use super::{
+    error::TransactionError,
+    objects::{internal_deploy::InternalDeploy, internal_invoke_function::InternalInvokeFunction},
+};
+
+pub(crate) enum Transaction {
+    Deploy(InternalDeploy),
+    InvokeFunction(InternalInvokeFunction),
+}
+
+impl Transaction {
+    pub fn contract_hash(&self) -> [u8; 32] {
+        match self {
+            Transaction::Deploy(tx) => tx.contract_hash,
+            _ => [0; 32],
+        }
+    }
```

Comment:
I remember that i took this decision because this executed before a validation that checks that the contract hash wasn't [0;32], so i thought it was convenient.